crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -4154,6 +4155,10 @@ impl<'db> TypeInferenceBuilder<'db> {
         collector.string_type(self.db())
     }
 
+    fn infer_tstring_expression(&mut self, _tstring: &ast::ExprTString) -> Type<'db> {
+        todo_type!("Template")
+    }
```

Comment:
To make the corpus test happy, you need to recurse inside all subnodes of the t-string and store a `todo_type!("Template")` for each sub-node. This is to support a future type-aware linter integration where the linter can query the type of any arbitrary node in the AST and ty will have an answer for it.